### DISCUSSION TITLE
Welcome to MyPTV Discussions!

### DISCUSSION THREAD

**Discussion by ronshnapp** [ronshnapp/MyPTV#25] [Announcements]
## 👋 Welcome!
  We’ll be using this Discussions forum as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

MyPTV is designed to be a community project for scientists to use in their own work and develop new features and ideas that will help advancing 3D particle tracking. We're excited to hear your thoughts and welcome your contributions.  

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zitaojiang1995** [comment]
Hello, 
I am a PhD student from Osaka University.
I just know MyPTV and this forum from Alex and OpenPTV forum.
I study natural ventilation and previously did particle track of natural ventilation in LES simualtion, which is very similar to PTV measurements but easier.
I recently know we can use PTV measurement for experiment, and read your papers about PTV measurement in wind tunnel (very impressive).
We are currently testing the PTV in indoor ventilation study and wind tunnel.
I hope this community can be more active and we can apply PTV in a variety of scenarios with accumulated knowledge and exeperience in this community.

**Reply by ronshnapp** [reply]
Hi @chinazjjzt, nice to meet you!
Your project sounds very interesting and I'm keen to learn more about it. Indeed, our project that you mentioned focused on the flow in between the canopy elements, so going forward and looking at the interaction of the outdoors-indoors through PTV is a great research topic.
In case you need help in setting up MyPTV let us know here, and we'll be glad to help out :)

**Comment by alexlib** [comment]
Is it possible to write a translator of the camera position from MyPTV to OpenPTV and vice versa? i.e. what are x,y,z -> x,y,z translations and angles -> angles rotations to position a camera, calibrated with MyPTV in the OpenPTV settings?

**Reply by ronshnapp** [reply]
Hi @alexlib. Yes, it should be possible, but might require a somewhat non-direct path. I think that the best thing would be to have a code that is independent of the optical model used in each software, since that will allow to continue developing the optical models and extend the translation of camera models to other software tools that we might want to integrate with in the future.

A possible way to do this would be to write a code that takes in a calibrated set of cameras in one softwares, generates synthetic particle data, and uses it to calibrate the cameras in the second software. How does this shound?

**Reply by alexlib** [reply]
sounds great. lets start with some boilerplate

**Reply by ronshnapp** [reply]
So as i see it, it shall be something like this: given two PTV softwares, we write a "translator" -

    translator( calibrated_cameras_in_software_1, method_for_generating_synthetic_data_in_software_1, calibration_method_for_software_2, format_for_writing_camera_files_in_software_2):

    .
    .
    .

    return calibrated_camera_files_in_spfware_2

**Reply by alexlib** [reply]
> So as i see it, it shall be something like this: given two PTV softwares, we write a "translator" -
> 
> 
> 
>     translator( calibrated_cameras_in_software_1, method_for_generating_synthetic_data_in_software_1, calibration_method_for_software_2, format_for_writing_camera_files_in_software_2):
> 
> 
> 
>     .
> 
>     .
> 
>     .
> 
> 
> 
>     return calibrated_camera_files_in_spfware_2 

ok now I see the direction. you mean to calibrate the camera in software 2 using the 3D positions generated by software 1 for the same images. 
interesting. need to think about it. 

isn't it easier to apply some rotation and translation between two coordinate systems?

**Reply by ronshnapp** [reply]
In principle yes, but this is only for the linear part of the optical model. I'm not sure there exists an easy transformation for the non-linear parts.

**Comment by jm013100** [comment]
Hello, @ronshnapp 
My name is Daniel, I am a master's student. 
I  know MyPTV and this forum from Alex and OpenPTV forum.
I am currently studying the track and distribution of particles in the spray. I would like to ask a few questions:
1. The MyPTV program can use an image taken by one camera to track particles? (In short, it is to detect particle displacement in 2D image)
2. At user_ In the manual.pdf file, I can't understand 2.2.3 Initial calibration - 1 The function of cam file generation. After I input values on the GUI interface, nothing appears on the Generate cam file button. I want to know its usage and how to use it more clearly.
Please answer the above questions, thanks you very much.

Best wishes

**Reply by ronshnapp** [reply]
Hi Daniel,

Nice to meet you!

1. Your question comes in a very good timing, because since I needed to have perform 2D tracking too, I extended MyPTV to include this feature. So, if you download the newest version of MyPTV (v 0.6.0 that I just uploaded here) you will be able to use MyPTV for your 2D tracking experiment. The instructions on how to do this are on Section 2.9 in the user manual.
2. Once you click the "Generate cam file" button the program generates a new file in which it saves the parameters of the camera and it saves it on your hard dive in the experiment folder. The name of the file is the name of the camera you used. It's okay that you don't see an indication of that on the GUI itself.  

Best,
Ron

**Reply by zitaojiang1995** [reply]
Hello Ronshanapp!
This is Jiang.
Thank you for including the 2D particle track components into the newest version of MyPTV.
I tested my 2D track experiment data and got stitiched trajectory txt file.
But I did not find the methods to visualization the results in the user manual, maybe it is written somewhere and I missed it?
I think the visualization can be done in Paraview , which is similar to OpenPTV.
Do we need to write a code for Paraview to read the tajectory?
Or are there any other methods?


Best regards,
Jiang

**Reply by ronshnapp** [reply]
Hi Jiang @chinazjjzt.,

I'm very glad to read that the 2D tracking worked for you!

At the moment we don't have a script or a general way for generating visualisations (although this is on my bucketlist). 
For now, I can recommend one of two ways to vizualize your results:
1) Using Python with Matplotlib. For that, you can load the results file (using e.g. Pandas), and plot the coordinates of the particles with a simple, either 2D or 3D plots in Matplotlib.
2) Using Paraview is indeed a great option too. For that, you could import the trajectory file as a .csv file format, apply threshold filters to choose any particular trajectories you might need, and finaly visualize the results by using the "table to points" filter.

Let me know if one of these works for you.

Best,
Ron
